test/jdk/javax/swing/JFileChooser/FileChooserSymLinkTest.java

```diff
@@ -0,0 +1,103 @@
+   @requires (os.family == "windows")
+   @library /java/awt/regtesthelpers
+   @build PassFailJFrame
+   @summary Test to check if the absolute path of Symbolic Link folder
+            is valid on ValueChanged property listener.
+   @run main/manual FileChooserSymLinkTest
+*/
+
+import java.awt.Dimension;
+import java.awt.Robot;
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.lang.reflect.InvocationTargetException;
+
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
+import javax.swing.WindowConstants;
+
+public class FileChooserSymLinkTest{
+    static JFrame frame = null;
+    static PassFailJFrame passFailJFrame = null;
+
+    public static void main(String[] args) throws Exception {
+        Robot robot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                try {
+                    initialize();
+                } catch (InterruptedException e) {
+                    throw new RuntimeException(e);
+                } catch (InvocationTargetException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+        });
+
+        robot.delay(2000);
+        robot.waitForIdle();
+
+        passFailJFrame.awaitAndCheck();
+    }
+
+    static void initialize() throws InterruptedException, InvocationTargetException {
+        final String INSTRUCTIONS = """
+                Instructions to Test:
+                1. Create a regular directory in any specific path.
+                    ex: mkdir c:\\target
+                2. Create a Symbolic link targeting the created test directory.
+                    ex : mklink /D c:\\link c:\\target
+                3. In JFileChooser, navigate to "link" created directed.
+                4. On click of the "link" directory, if the Absolute path of
+                    Symbolic Link is valid then Click PASS, else if it is
+                    null then Click FAIL.
+                """;
+        frame = new JFrame("JFileChooser Symbolic Link test");
+        passFailJFrame = new PassFailJFrame(INSTRUCTIONS);
+        PassFailJFrame.addTestWindow(frame);
+        PassFailJFrame.positionTestWindow(frame, PassFailJFrame.Position.HORIZONTAL);
+
+        frame.setPreferredSize(new Dimension(600, 600));
+        frame.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
+        JFileChooser jfc = new JFileChooser();
+        jfc.setDialogType(JFileChooser.CUSTOM_DIALOG);
+        jfc.setControlButtonsAreShown(false);
+        jfc.setFileSelectionMode(JFileChooser.DIRECTORIES_ONLY);
+        jfc.addPropertyChangeListener(new PropertyChangeListener() {
+            public void propertyChange(PropertyChangeEvent evt) {
+                if (JFileChooser.SELECTED_FILE_CHANGED_PROPERTY.equals(evt.getPropertyName())) {
+                    System.out.println(String.format("Absolute Path : %s",evt.getNewValue()));
```

Comment:
When with jtreg, the output to the console is not seen.

You have to provide a UI where these lines would be added.

The usage of `format` for such a simple case could be replaced by string concatenation using `+`.

In addition to that, you may keep track whether a `null` value is seen or not. No `null` value is expected. If it's seen, it's a failure.